olp-cpp-sdk-authentication/src/AuthenticationClientImpl.cpp

```diff
@@ -555,81 +555,46 @@ client::CancellationToken AuthenticationClientImpl::HandleUserRequest(
 client::CancellationToken AuthenticationClientImpl::SignUpHereUser(
     const AuthenticationCredentials& credentials,
     const SignUpProperties& properties, const SignUpCallback& callback) {
-  if (!settings_.network_request_handler) {
-    thread::ExecuteOrSchedule(settings_.task_scheduler, [callback] {
-      callback(
-          client::ApiError::NetworkConnection("Cannot sign up while offline"));
-    });
-    return client::CancellationToken();
-  }
-  std::weak_ptr<http::Network> weak_network(settings_.network_request_handler);
-  std::string url = settings_.token_endpoint_url;
-  url.append(kUserEndpoint);
-  http::NetworkRequest request(url);
-  auto network_settings =
-      http::NetworkSettings()
-          .WithTransferTimeout(settings_.retry_settings.timeout)
-          .WithConnectionTimeout(settings_.retry_settings.timeout)
-          .WithProxySettings(settings_.network_proxy_settings.get_value_or({}));
+  using ResponseType = client::ApiResponse<SignUpResult, client::ApiError>;
 
-  request.WithVerb(http::NetworkRequest::HttpVerb::POST);
+  auto signup_task = [=](client::CancellationContext context) -> ResponseType {
+    if (!settings_.network_request_handler) {
+      return client::ApiError::NetworkConnection(
+          "Cannot sign up while offline");
+    }
 
-  auto auth_header = GenerateAuthorizationHeader(
-      credentials, url, std::time(nullptr), GenerateUid());
+    if (context.IsCancelled()) {
```

Comment:
Do we need the part? This will automatically happen inside CallApi?

Reply:
I guess why bother setting up the auth headers and all the stuff if it gets dropped the next line.

Reply:
If the task is canceled, we are exiting directly from the TaskContext, else we exit from CallApi

Reply:
Yes, but there is a small chance that the cancellation hits us right after the check in TaskContext.